plonetheme/onegovbear/browser/dynamic_scss_resources.py

```diff
@@ -0,0 +1,59 @@
+from Acquisition._Acquisition import aq_inner
+from Acquisition._Acquisition import aq_parent
+from Products.CMFPlone.interfaces import IPloneSiteRoot
+from ftw.theming.interfaces import ISCSSResourceFactory
+from ftw.theming.resource import SCSSResource
+from plone.app.layout.navigation.interfaces import INavigationRoot
+from plonetheme.onegovbear.browser.forms import ANNOTATION_KEY
+from zope.annotation import IAnnotations
+from zope.interface import implements
+
+
+class CustomVariablesResourceFactory(object):
+    implements(ISCSSResourceFactory)
+    annotations_key = ANNOTATION_KEY
+
+    def __call__(self, context, request):
+        self.context = context
+        self.request = request
+        return self.get_resource()
+
+    def get_resource(self):
+        source = self.get_source()
+        return SCSSResource(ANNOTATION_KEY, slot='variables', source=source)
+
+    def get_source(self):
+        ancestor_variables = self.get_ancestor_variables()
+
+        variables = {}
+        for d in ancestor_variables:
+            variables.update(d)
+
+        source = ''
+        if variables:
+            source = '; '.join(
+                ['{0}: {1}'.format(value['variable_name'], value['value'])
+                 for value in variables.itervalues()]
+            ) + ';'
+        return source
+
+    def get_ancestor_variables(self):
+        obj = self.context
+        ancestor_variables = []
+        while True:
```

Comment:
Just as idea: maybe it is simpler using `aq_chain` here:
```python
>>> from Acquisition import aq_chain
>>> aq_chain(obj)
[<ContentPage at /platform/themen/abfall/kehrichtabfuhr>, <ContentPage at /platform/themen/abfall>, <ContentPage at /platform/themen>, <PloneSite at /platform>, <Application at >, <ZPublisher.BaseRequest.RequestContainer object at 0x10e1f4ad0>]
```

You could do something like:
```python
ancestor_variables  = filter(bool, map(self.get_scss_variables, aq_chain(self.context)))
```